docs/_posts/2022-10-31-luau-semantic-subtyping.md

```diff
@@ -0,0 +1,239 @@
+---
+layout: single
+title:  "Semantic Subtyping in Luau"
+author: Alan Jeffrey
+---
+
+Luau is the first production language to include semantic subtyping, which removes one source of spurious type errors.
+
+## Minimizing false positives
+
+One of the issues with type error reporting in tools like the Script Analysis widget in Roblox Studio is *false positives*. These are warnings that are artifacts of the analysis, and don’t correspond to errors which can occur at runtime. For example, the program
+```lua
+  local x
+  x = nil
+  x = 5
+  print(x + 1)
+```
+reports a type error which cannot happen at runtime, since by the time `x + 1` is executed, `x` is `5`, not `nil`.
+
+False positives are especially poor for onboarding new users. If a type-curious creator switches on typechecking and is immediately faced with a wall of spurious red squiggles, there is a strong incentive to immediately switch it off again.
+
+Inaccuracies in type errors are inevitable, since it is undecidable to decide ahead of time whether a runtime error will be triggered. Type system designers have to choose whether to live with false positives or false negatives. In Luau this is determined by the mode: `strict` mode errs on the side of false positives, and `nonstrict` mode errs on the side of false negatives.
+
+While inaccuracies are inevitable, we try to remove them whenever possible, since they result in spurious errors, and imprecision in type-driven tooling like autocorrect or API documentation.
```

Comment:
```suggestion
While inaccuracies are inevitable, we try to remove them whenever possible, since they result in spurious errors, and imprecision in type-driven tooling like autocomplete or API documentation.
```